src/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -831,15 +844,13 @@ export const ShippingRoute: FC<ShippingProps> = props => {
                     setAddressNeedsVerification(false)
                     setAddressVerifiedBy(AddressVerifiedBy.USER)
                   }}
-                  onChosenAddress={(
-                    verifiedBy,
-                    chosenAddress,
-                    saveAndContinue
-                  ) => {
+                  onChosenAddress={(verifiedBy, chosenAddress) => {
                     setAddressNeedsVerification(false)
                     setAddressVerifiedBy(verifiedBy)
-                    setAddress({ ...address, ...chosenAddress })
-                    saveAndContinue && finalizeFulfillment()
+                    setAddress(address => {
+                      return { ...address, ...chosenAddress }
+                    })
+                    setReadyToSaveVerifiedAddress(true)
```

Comment:
Can we add a comment above this line describing exactly what happens in terms of triggering the button in the effect? That way there's no ambiguity.